python/tskit/trees.py

```diff
@@ -2209,6 +2209,15 @@ def nodes(self, root=None, order="preorder"):
             for u in roots:
                 yield from iterator(u)
 
+    def _node_edges(self):
+        """
+        Return a numpy array mapping the node IDs in this tree to the ID of the edge
+        above them. This is in lieu of a tree.edge(u) function, currently implemented
+        using the non-optimised Python TreeSequence._tree_node_edges() generator
+        """
+        ts = self.tree_sequence
+        return next(x for i, x in enumerate(ts._tree_node_edges()) if i == self.index)
```

Comment:
This means we always iterate over all trees, right? How about
```python
for index, node_edge_map in enumerate(self.tree_sequence._tree_node_edges()):
    if index == self.index:
          return node_edge_map
```
It's a bit more obvious as well

Reply:
Ah, maybe it doesn't, next will pop the first thing off the iterator. Anyway, I shouldn't have to think this hard to figure out what this is doing, it should be immediately obvious.

Reply:
Yeah, it's a generator - method picked straight off Stack Overflow. But easy to switch to your version.